launcher/minecraft/UpdateGlobalDirectoriesTask.cpp

```diff
@@ -0,0 +1,182 @@
+#include "UpdateGlobalDirectoriesTask.h"
+
+#include <QDirIterator>
+
+#include "Application.h"
+#include "FileSystem.h"
+#include "minecraft/MinecraftInstance.h"
+#include "tasks/ConcurrentTask.h"
+#include "ui/dialogs/CustomMessageBox.h"
+#include "ui/dialogs/FileConflictDialog.h"
+
+/**
+ * @brief Move a file or folder, and ask the user what to do in case of a conflict.
+ * @param source      What to move.
+ * @param destination Where to move it to.
+ * @param recursive   If true, all direct children will be moved 1 by 1.
+ *                    If false, the source will be directly moved to the destination.
+ * @param parent      The parent of the dialog.
+ * @return True if everything could be moved.
+ */
+bool interactiveMove(const QString& source, const QString& destination, bool recursive = false, QWidget* parent = nullptr)
+{
+    const QFileInfo sourceInfo(source);
+
+    // Make sure the source exists.
+    if (!sourceInfo.exists())
+        return false;
+
+    if (recursive) {
+        // Recursive doesn't make sense if the source isn't a directory.
+        if (!sourceInfo.isDir())
+            return false;
+
+        QDirIterator sourceIt(source, QDir::Filter::Files | QDir::Filter::Dirs | QDir::Filter::Hidden | QDir::Filter::NoDotAndDotDot);
+
+        while (sourceIt.hasNext()) {
+            if (!interactiveMove(sourceIt.next(), FS::PathCombine(destination, sourceIt.fileName()), false))
+                return false;
+        }
+
+        return true;
+    }
```

Comment:
```suggestion
    if (recursive && sourceInfo.isDir()) {
        // Recursive doesn't make sense if the source isn't a directory.

        QDirIterator sourceIt(source, QDir::Filter::Files | QDir::Filter::Dirs | QDir::Filter::Hidden | QDir::Filter::NoDotAndDotDot);

        while (sourceIt.hasNext()) {
            if (!interactiveMove(sourceIt.next(), FS::PathCombine(destination, sourceIt.fileName()), false))
                return false;
        }

        return true;
    }
```


IMO it makes much more sense to just ignore the recursion param if the source isn't a directory